api/maven-api-model/src/main/java/org/apache/maven/api/model/InputSource.java

```diff
@@ -64,6 +74,14 @@ public String getModelId() {
         return this.modelId;
     }
 
+    public InputLocation getImportedFrom() {
+        return importedFrom;
+    }
+
+    public InputSource importedFrom(InputLocation importedFrom) {
+        return new InputSource(modelId, location, importedFrom);
```

Comment:
really create a new object and forget about it? strange (i did not really dig in depth, just giving initial feedback)

Reply:
This is due to the immutable model, but I think we can simply remove this and make the new constructor public. We don't seem to use it (anymore) upon further inspection...